test/extensions/filters/http/cache/cacheability_utils_test.cc

```diff
@@ -74,6 +76,17 @@ TEST_F(IsCacheableRequestTest, AuthorizationHeader) {
   EXPECT_FALSE(CacheabilityUtils::isCacheableRequest(request_headers));
 }
 
+INSTANTIATE_TEST_SUITE_P(ConditionalHeaders, IsCacheableRequestTest,
+                         testing::Values("if-match", "if-none-match", "if-modified-since",
+                                         "if-unmodified-since", "if-range"));
+
+TEST_P(IsCacheableRequestTest, ConditionalHeaders) {
+  Http::TestRequestHeaderMapImpl request_headers = cacheable_request_headers_;
+  EXPECT_TRUE(CacheabilityUtils::isCacheableRequest(request_headers));
+  request_headers.setCopy(Http::LowerCaseString{GetParam()}, "test-value");
```

Comment:
Just to make sure I understand... We don't care what format the If-Modified-Since (or If-Unmodified-Since) dates are in and we just proxy them through to the backend, and will use whatever the request gives us?

Reply:
According to: https://httpwg.org/specs/rfc7232.html#header.if-modified-since we can also use the `date` field or any other date that makes sense. I think we should modify the functionality to fallback to the date if `last-modified` is not valid or non-existent, and add a test for this. Will work on that.

Reply:
I just realized what part of the code this comment is on. This test only checks that if a conditional header is present then the request bypasses the Cache.
However your comment made me realize the possibility of using `Date` instead of `Last-Modified` for validation if `Last-Modified` is non-existent or ill-formed (which I already added).

Reply:
Cool. So `test-value` doesn't really matter for the date formatted fields cause we don't actually parse it ever.

Reply:
Yes. If the request already contains conditional headers it should pass the CacheFilter regardless of its value.